### PR TITLE
feat(Task-list-scrolling): enable auto-scrolling to newest task addit…

### PR DESCRIPTION
**Pull request by rrrekin**
…ions and improve task list styles

- Added `scrollToBottom` functionality to `TaskList` for smooth scrolling on task additions.
- Updated styles to auto-hide scrollbars on task list hover.
- Adjusted template ref bindings in `App.vue`.
- Bumped Electron and other dependencies in `package.json` and `package-lock.json`.

## Summary by CodeRabbit

- New Features
  - Task list auto-scrolls when tasks are added, replayed, or loaded; categories list auto-scrolls to show the add form; app now handles app version retrieval gracefully.

- Style
  - Enabled vertical scrolling with refined auto-hide and platform-specific scrollbar behavior for the task table pane.

- Tests
  - Standardized test formatting, added scroll-behavior tests and extensive App branch/edge-case coverage, and improved test cleanup.

- Chores
  - Electron bumped to 38.0.0; Node engine set to >=20; CI/install adjusted to optionally skip/rebuild native modules.

- Documentation
  - Documented environment variable to skip native rebuild during install.

### CODE REVIEW

**Comment by rrrekin** [comment]
@coderabbitai review

File: src/components/TaskList.vue [hunk 1]
```diff
@@ -249,6 +249,25 @@ const formDropdownTriggerId = `form-dropdown-trigger-${componentId}`
...
+
+  const wrapper = el.closest('.task-table-pane') as HTMLElement | null
+  const container = (wrapper ?? el) as HTMLElement
+
+  if (typeof (container as any).scrollTo === 'function') {
+    container.scrollTo({ top: container.scrollHeight, behavior })
+  } else {
+    ;(container as any).scrollTop = container.scrollHeight
+  }
+}
```
[the remark this answers is not in the document]
Comment (rrrekin) [reply src/components/TaskList.vue:269]: This intentional, the user experience should be that the pane is immediately ready to accept new entry.
